tests/integration/records/relationship-changes-test.js

```diff
@@ -949,3 +950,112 @@ test('Calling push with same belongsTo relationship does not trigger observer',
 
   assert.equal(observerCount, 0, 'author observer should not be triggered');
 });
+
+test('Calling push with hasMany relationship does trigger relation observer with already loaded objects', function(assert) {
```

Comment:
you're not pushing a relationship - you're pushing an object that changes the inverse hasMany